src/GraphicsView.cpp

```diff
@@ -180,17 +181,39 @@ wheelEvent(QWheelEvent *event)
 }
 
 
+void
+GraphicsView::
+setScaleRange(double min, double max)
+{
+    if(min <= max)
+    {
+        _scaleRange.setX(min < 0 ? 0 : min);
+        _scaleRange.setY(max < 0 ? 0 : max);
+    }
+    else
+    {
+        _scaleRange.setX(max < 0 ? 0 : max);
+        _scaleRange.setY(min < 0 ? 0 : min);
+    }
+}
```

Comment:
1. The function should check the current scale of the scene and adjust it if we are out of range.
2. Indentation is currently 2 whitespaces.